chrome-habit-tracker-master/index.js

```diff
@@ -0,0 +1,222 @@
+    habitContainerNode.addEventListener('mouseout', hideRemove);
+    const removeNode = createNode('span', 'remove-habit');
+    removeNode.setAttribute('id', `remove-${i + 1}`);
+    removeNode.addEventListener('click', removeHabit);
+    const removeTextNode = document.createTextNode('X');
+    removeNode.appendChild(removeTextNode);
+    document.getElementById(`habit-container-${i + 1}`).appendChild(removeNode);
+    const habitNode = createNode('span', 'habit');
+    const textNode = document.createTextNode(habitStorage.habits[i].name);
+    habitNode.appendChild(textNode);
+    habitNode.setAttribute('id', `habit-${i + 1}`);
+    document.getElementById(`habit-container-${i + 1}`).appendChild(habitNode);
+    for (let j = 0; j < actualDays; j++) {
+      const checkNode = createNode('input', 'check');
+      checkNode.setAttribute('id', `checkbox-${i + '-' + j}`);
+      checkNode.setAttribute('type', 'checkbox');
+      checkNode.addEventListener('click', toggleCheck);
+      const labelCheckNode = document.createElement('label');
+      labelCheckNode.setAttribute('for', `checkbox-${i + '-' + j}`);
+      const checkContainerNode = createNode('div', 'check-container');
+      checkContainerNode.appendChild(checkNode);
+      checkContainerNode.appendChild(labelCheckNode);
+      if (j === actualDays - 1) checkContainerNode.classList.add('last-one');
+      document.getElementById('grid-main-container').appendChild(checkContainerNode);
+      // input value logic: HABIT ID 2 DIGITS | YEAR 4 DIGITS | MONTH 2 DIGITS | DAY 2 DIGITS
+      const dateValue = `${String(habitStorage.habits[i].id)}${new Date().getFullYear()}${monthSelected < 10 ? '0' + String(monthSelected + 1) : monthSelected + 1}${j + 1 < 10 ? '0' + String(j + 1) : String(j + 1)}`;
+      const dateValueIdLessThan10 = `0${dateValue}`;
+      checkNode.setAttribute('value', `${habitStorage.habits[i].id < 10 ? dateValueIdLessThan10 : dateValue}`);
+      if (habitStorage.habits[i].dates.hasOwnProperty(checkNode.value) && habitStorage.habits[i].dates[checkNode.value] === 1) {
+        checkNode.setAttribute('checked', true);
+      }
+      checkNode.addEventListener('click', toggleCheck);
+      document.getElementById('grid-main-container').appendChild(checkContainerNode);
+    }
+  }
+}
+
+const toggleCheck = (e) => {
+  const habitClicked = habitStorage.habits[Number(e.target.value.slice(0, 2)) - 1]; // habits arr is 0 - indexed and habits id are 1 - indexed
+  if (habitClicked.dates.hasOwnProperty(e.target.value)) {
+    if (habitClicked.dates[e.target.value] === 0) {
+      habitClicked.dates[e.target.value] = 1;
+    } else {
+      habitClicked.dates[e.target.value] = 0;
+    }
+  } else {
+    habitStorage.habits[Number(e.target.value.slice(0, 2)) - 1].dates[e.target.value] = 1;
+  }
+  chrome.storage.sync.set(habitStorage, () => console.log('Check toggled'));
+}
+
+const showInput = () => {
+  document.getElementById('add-habit-container').style['display'] = 'none';
```

Comment:
when doing DOM updates take this into account :

https://stackoverflow.com/questions/37039667/executing-multiple-dom-updates-with-javascript-efficiently